### PR TITLE
Validate schema of events after test is complete

### PR DESCRIPTION
**Pull request by mennopruijssers**
Before this PR, we were validating each event when it happens. Especially on slower machines (e.g. travis), the impact is severe enough to interfere with some tests. Local testing showed that (with cpu throttled to 5%), event validation takes over 2.5 seconds for the bidirectional full sync tests.

### CODE REVIEW

File: test/ringpop-assert.js [hunk 1]
```diff
@@ -791,11 +790,10 @@ function createValidateEvent(t, tc) {
 function validate(t, tc, scheme, deadline) {
     var fns = scheme;
     var cursor = 0;
+    var allEvents = [];
     var eventList = [];
 
-    tc.on('event', createValidateEvent(t, tc));
```
Comment (mennopruijssers) [inline test/ringpop-assert.js]: Validation is moved to line 866

File: test/ringpop-assert.js [hunk 2]
```diff
@@ -824,12 +822,7 @@ function validate(t, tc, scheme, deadline) {
         inProgress = true;
 
         if(cursor >= fns.length) {
-            clearTimeout(timer);
-            t.ok(true, 'validate done: all functions passed');
-            tc.shutdown();
-            tc.removeAllListeners('event');
-            tc.removeAllListeners('sut-died');
-            t.end();
+            progressionComplete();
```
Comment (mennopruijssers) [inline test/ringpop-assert.js]: These lines are extracted in the `progressionComplete`-function.

File: test/ringpop-assert.js [hunk 3]
```diff
@@ -858,8 +851,23 @@ function validate(t, tc, scheme, deadline) {
 
     tc.on('event', function(event) {
         eventList.push(event);
+        allEvents.push(event);
```
Comment (mennopruijssers) [inline test/ringpop-assert.js]: We need a second list, because events that are handled by the test are removed from the `eventList`.

**Comment by severb** [comment]
LGTM!

**Pull request merged**